redis/src/aio.rs

```diff
@@ -679,26 +692,37 @@ where
 
     fn send_result(self: Pin<&mut Self>, result: Result<I, E>) {
         let self_ = self.project();
-        let response = {
+
+        {
             let entry = match self_.in_flight.front_mut() {
                 Some(entry) => entry,
                 None => return,
             };
+
             match result {
                 Ok(item) => {
                     entry.buffer.push(item);
-                    if entry.response_count > entry.buffer.len() {
-                        // Need to gather more response values
-                        return;
+                }
+                Err(err) => {
```

Comment:
So do we typically expect other return items to come in even after an error?

Reply:
Yes. I will admit that intuitively it's a bit scary, but this is how the other `Connection`s (e.g., for [aio::Connection](https://github.com/redis-rs/redis-rs/blob/d7d2f2a6b868ba6e4527110c1dd170d3675cb5c1/redis/src/aio.rs#L572), and for [the sync Connection](https://github.com/redis-rs/redis-rs/blob/d7d2f2a6b868ba6e4527110c1dd170d3675cb5c1/redis/src/connection.rs#L872)) and indeed the other redis libraries I've looked at handle the situation. Unless there's an issue parsing the commands properly, or a larger infrastructure issue, the commands will all be sent and responses will be returned.

Again, I think the API itself here is poor. We really should return `Result`s for individual commands rather than all-or-nothing but this fix needs to happen irrespective of that.

Reply:
> We really should return Results for individual commands rather than all-or-nothing but this fix needs to happen irrespective of that.

I actually was looking for a FromRedisValue implementation for Result for this reason 😄 

You might consider adding that as an option rather than pushing Result onto every consumer of your pipe api? Fwiw it's nice to be able to assert that the whole thing worked _else_ it is an error. But for some use cases a partial success is important and may be distinct from an overall pipe failure - hence, a FromRedisValue<Result...> impl which would even let someone decide if 1 part of the pipeline is resulty distinct from the overall pipe!

Reply:
PR welcome? 😄